File: articles/post-jungian-developments-in-clinical-analytical-psychology/c/clinical_adequacy.c
/*
Post-Jungian Developments in Clinical Analytical Psychology
C example: simple synthetic clinical-adequacy score

Synthetic educational demonstration only.
Not for diagnosis, treatment assignment, assessment, screening,
surveillance, or individual evaluation.
*/

#include <stdio.h>

double clinical_adequacy(
    double symbolic_depth,
    double relational_sophistication,
    double developmental_precision,
    double trauma_sensitivity,
    double embodied_regulation,
    double cultural_responsiveness,
    double doctrinal_rigidity
) {
    return 0.50 * symbolic_depth
         + 0.62 * relational_sophistication
         + 0.58 * developmental_precision
         + 0.66 * trauma_sensitivity
         + 0.52 * embodied_regulation
         + 0.46 * cultural_responsiveness
         - 0.60 * doctrinal_rigidity;
}

int main(void) {
    double score = clinical_adequacy(0.84, 0.86, 0.82, 0.86, 0.82, 0.78, 0.24);
    printf("Synthetic clinical-adequacy score: %.3f\n", score);
    return 0;
}

File: articles/post-jungian-developments-in-clinical-analytical-psychology/fortran/clinical_adequacy.f90
! Post-Jungian Developments in Clinical Analytical Psychology
! Fortran example: simple synthetic clinical-adequacy score

program clinical_adequacy_demo
  implicit none

  real :: score

  score = 0.50 * 0.84 + 0.62 * 0.86 + 0.58 * 0.82 + &
          0.66 * 0.86 + 0.52 * 0.82 + 0.46 * 0.78 - &
          0.60 * 0.24

  print *, "Synthetic clinical-adequacy score:", score
end program clinical_adequacy_demo
